src/Engine/EngineBase.php

```diff
@@ -7,26 +7,45 @@
 
 abstract class EngineBase
 {
+
     public const ALLOWED_FORMATS = ['png', 'jpeg', 'jpg', 'gif', 'tiff', 'tif', 'webp'];
 
+    /** @var string[] The host names for the images. */
+    protected $imageHosts = [];
+
     /**
      * @param string $imageUrl
      * @param string[]|null $langs
      * @return string
      */
     abstract public function getText(string $imageUrl, ?array $langs = null): string;
 
+    public function setImageHosts(string $imageHosts): void
+    {
+        $this->imageHosts = array_map('trim', explode(',', $imageHosts));
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getImageHosts(): array
+    {
+        return $this->imageHosts;
+    }
+
     /**
      * Checks that the given image URL is valid.
      * @param string $imageUrl
      * @throws OcrException
      */
     public function checkImageUrl(string $imageUrl): void
     {
+        $hostRegex = implode('|', array_map('preg_quote', $this->getImageHosts()));
         $formatRegex = implode('|', self::ALLOWED_FORMATS);
-        $matches = preg_match("/^https?:\/\/upload\.wikimedia\.org\/.*($formatRegex)$/", strtolower($imageUrl));
+        $regex = "/https?:\/\/($hostRegex).*($formatRegex)$/";
+        $matches = preg_match($regex, strtolower($imageUrl));
         if (1 !== $matches) {
-            throw new OcrException('image-url-error', ['https://upload.wikimedia.org']);
+            throw new OcrException('image-url-error', [join(', ', $this->getImageHosts())]);
```

Comment:
This should also use `Intuition::listToText()` but I'm not quite sure what the best way to do it is.

It's probably easiest to just give EngineBase a Intuition member variable, but that feels slightly hacky.

Reply:
I injected the Intuition member variable in https://github.com/wikimedia/wikimedia-ocr/pull/23/files so if that gets approved then you can just use the `Intuition::listToText()` :)